### PR TITLE
Fix for Imviz parser romandeps test after update to defaults

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

In #2767 I corrected the ASDF extensions loaded by default in the Imviz parser. The optional Imviz parser tests for Roman data products hadn't been updated to match. I fixed them in this PR.

This PR needs the `Extra CI` label to run the Roman tests. That means the link check also gets run on this PR, so I've also discovered and fixed some broken links in this same PR.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/sample_notebooks.rst [hunk 1]
```diff
@@ -10,6 +10,6 @@ These tools include three types of example Jupyter notebooks.
 
 1. :gh-tree:`Sample Jdaviz Notebooks <notebooks>` that illustrate how to use Jdaviz and various API calls. These notebooks are located in the ``notebooks`` sub-directory of the git repository.
 
-2. `Sample JDAT notebooks <https://jwst-docs.stsci.edu/jwst-post-pipeline-data-analysis/data-analysis-example-jupyter-notebooks>`_ that illustrate likely science workflows with data obtained from the various JWST instruments. These notebook incorporate ``astropy`` and Jdaviz when possible.
+2. `Sample JDAT notebooks <https://jwst-docs.stsci.edu/jwst-post-pipeline-data-analysis#JWSTPostPipelineDataAnalysis-JWSTdataanalysis(JDAT)notebooks>`_ that illustrate likely science workflows with data obtained from the various JWST instruments. These notebook incorporate ``astropy`` and Jdaviz when possible.
```
Comment (kecnry) [inline docs/sample_notebooks.rst:13]: while we're at it...

```suggestion
2. `Sample JDAT notebooks <https://jwst-docs.stsci.edu/jwst-post-pipeline-data-analysis#JWSTPostPipelineDataAnalysis-JWSTdataanalysis(JDAT)notebooks>`_ that illustrate likely science workflows with data obtained from the various JWST instruments. These notebooks incorporate ``astropy`` and Jdaviz when possible.
```

File: jdaviz/configs/imviz/tests/test_parser_roman.py [hunk 2]
```diff
@@ -8,18 +8,20 @@
...
+    [(None, 1),
      ('data', 1),
      (['data', 'var_rnoise'], 2)])
 def test_roman_wfi_ext_options(imviz_helper, roman_imagemodel, ext_list, n_dc):
     imviz_helper.load_data(roman_imagemodel, data_label='roman_wfi_image_model', ext=ext_list)
     dc = imviz_helper.app.data_collection
     assert len(dc) == n_dc
 
-    if ext_list is None:
+    if ext_list == '*':
```
Comment (kecnry) [inline jdaviz/configs/imviz/tests/test_parser_roman.py:19]: should this be covered in the test matrix?

**Review by kecnry — APPROVED** [review]
Minor comments - feel free to ignore or address and merge

File: jdaviz/configs/imviz/tests/test_parser_roman.py (hunk 2 above)
Comment (bmorris3) [reply jdaviz/configs/imviz/tests/test_parser_roman.py:19]: I'm going to say let's worry about coverage here once we have functionality that requires this behavior.

**Comment by bmorris3** [comment]
Remaining failures unrelated, merging. Thanks!

**Pull request merged**